core/services/keystore/keystoretest.go

```diff
@@ -0,0 +1,80 @@
+package keystore
+
+import (
+	"sync"
+
+	"github.com/smartcontractkit/sqlx"
+
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/pg"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+// memoryORM is an in-memory version of the keystore. This is
+// only intended to be used in tests to avoid DB lock contention on
+// the single DB row that stores the key material.
+//
+// Note: we store `q` on the struct since `saveEncryptedKeyRing` needs
+// to support DB callbacks.
+type memoryORM struct {
+	keyRing *encryptedKeyRing
+	q       pg.Queryer
+	mu      sync.RWMutex
+}
+
+func (o *memoryORM) isEmpty() (bool, error) {
+	return false, nil
+}
+
+func (o *memoryORM) saveEncryptedKeyRing(kr *encryptedKeyRing, callbacks ...func(pg.Queryer) error) error {
+	o.mu.Lock()
+	defer o.mu.Unlock()
+	o.keyRing = kr
+	for _, c := range callbacks {
+		c(o.q)
```

Comment:
@chainchad the linter didn't flag on this unchecked error, but it did come up in a local run :disappointed: